packages/sdk-communication-layer/src/services/ConnectionManager/initCommunicationLayer.ts

```diff
@@ -0,0 +1,135 @@
+ *
+ * If the dappMetadata is available, this metadata is used to populate originator information such as the URL and name of the dapp.
+ * This function also sets up various event listeners to handle different types of events that can occur in the communication layer, ensuring that the RemoteCommunication instance responds appropriately to each event.
+ *
+ * @param communicationLayerPreference Specifies the preferred communication protocol (e.g., SOCKET).
+ * @param otherPublicKey The public key of the other party for communication (if available).
+ * @param reconnect Indicates if the communication layer should attempt to reconnect after disconnection.
+ * @param ecies The Elliptic Curve Integrated Encryption Scheme properties.
+ * @param communicationServerUrl The URL of the communication server, defaults to the value in DEFAULT_SERVER_URL.
+ * @param instance The current instance of the RemoteCommunication class.
+ * @throws Error when an invalid communication protocol is specified.
+ */
+export function initCommunicationLayer({
+  communicationLayerPreference,
+  otherPublicKey,
+  reconnect,
+  ecies,
+  communicationServerUrl = DEFAULT_SERVER_URL,
+  instance,
+}: {
+  communicationLayerPreference: CommunicationLayerPreference;
+  otherPublicKey?: string;
+  reconnect?: boolean;
+  ecies?: ECIESProps;
+  communicationServerUrl?: string;
+  instance: RemoteCommunication;
+}) {
+  // instance.state.communicationLayer?.removeAllListeners();
+
+  switch (communicationLayerPreference) {
+    case CommunicationLayerPreference.SOCKET:
+      instance.state.communicationLayer = new SocketService({
+        communicationLayerPreference,
+        otherPublicKey,
+        reconnect,
+        transports: instance.state.transports,
+        communicationServerUrl,
+        context: instance.state.context,
+        ecies,
+        logging: instance.state.logging,
+      });
+      break;
+    default:
+      throw new Error('Invalid communication protocol');
+  }
+
+  let url = (typeof document !== 'undefined' && document.URL) || '';
+  let title = (typeof document !== 'undefined' && document.title) || '';
+
+  if (instance.state.dappMetadata?.url) {
+    url = instance.state.dappMetadata.url;
+  }
+
+  if (instance.state.dappMetadata?.name) {
+    title = instance.state.dappMetadata.name;
+  }
+
+  const originatorInfo: OriginatorInfo = {
+    url,
+    title,
+    source: instance.state.dappMetadata?.source,
+    icon: instance.state.dappMetadata?.base64Icon,
+    platform: instance.state.platformType,
+    apiVersion: packageJson.version,
+  };
+  instance.state.originatorInfo = originatorInfo;
+
+  // TODO below listeners is only added for backward compatibility with wallet < 7.3
+  instance.state.communicationLayer?.on(
```

Comment:
what about doing something more compact and clean like:
```
const { communicationLayer } = instance.state;

if (!communicationLayer) {
  return; // Exit early if communicationLayer is not defined
}

const eventsMapping: { [key in EventType]: (...args: any[]) => void } = {
  [EventType.AUTHORIZED]: () => handleAuthorizedEvent(instance),
  [EventType.MESSAGE]: () => handleMessageEvent(instance),
  [EventType.CLIENTS_CONNECTED]: () => handleClientsConnectedEvent(instance, communicationLayerPreference),
  [EventType.KEYS_EXCHANGED]: () => handleKeysExchangedEvent(instance, communicationLayerPreference),
  [EventType.SOCKET_DISCONNECTED]: () => handleSocketDisconnectedEvent(instance),
  [EventType.SOCKET_RECONNECT]: () => handleSocketReconnectEvent(instance),
  [EventType.CLIENTS_DISCONNECTED]: handleClientsDisconnectedEvent, // Assuming this doesn't need any parameters
  [EventType.CHANNEL_CREATED]: () => handleChannelCreatedEvent(instance),
  [EventType.CLIENTS_WAITING]: () => handleClientsWaitingEvent(instance),
};

for (const [eventType, handler] of Object.entries(eventsMapping)) {
  communicationLayer.on(eventType, handler);
}
```

Reply:
Looks great, I have changed that :)